doc/controller.md

```diff
@@ -0,0 +1,132 @@
+
+## Database Verification
+
+Database verification is currently done on startup (or a restart) within the same backup-restore sidecar's main process.
+There is a co-ordination shell script that makes sure that the etcd process is not started until verification is completed successfully.
+
+Introducing a controller enables the option to perform the database verification only when needed as a separate job.
+This has two advantages.
+1. Decouple database verification from `etcd` restart. This has the potential to avoid unnecessary delays during every single `etcd` restart.
+1. Decouple the memory requirement of a database verification from the regular backup (full and delta) tasks. This has the potential to reduce the memory requirement of the backup sidecar and isolate the memory spike of a database verification.
+
+Of course, it is possible to decouple database verification from `etcd` restart without introducing the controller but would need the co-ordination shell script to be more complicated as can be seen in [#93](https://github.com/gardener/etcd-backup-restore/pull/93). Such complicated shell scripts are generally better avoided. Especially, when they are not even part of any docker image.
+
+Another alternative is to create a custom image for etcd container to include the co-ordination logic. This is also not very desirable.
+
+As mentioned in the case of [database restoration](#database-restoration), database verification is a part of a subcommand of the existing backup-restore component. So, the existing verification functionality can be fully reused even at the level of the binary docker image.
+
+## Autoscaling
+
+The [`VerticalPodAutoscaler`](https://github.com/kubernetes/autoscaler/tree/master/vertical-pod-autoscaler) supports multiple update policies including `recreate`, `initial` and `off`.
+The `recreate` policy is clearly not suitable for a single-node `etcd` instances because of the implications on frequent and unpredictable down-time.
+The `initial` policy makes more sense when coupled with unlimited resource limits (but very clear autoscaled resource requests).
+
+With a controller, even the `off` option becomes feasible because the time for applying `VerticalPodAutoscaler`'s could be decided in a very custom way while still relying on the recommendations from the `VerticalPodAutoscaler`.
+
+## Non-disruptive Maintenance
+
+We currently support only single-node `etcd` instances.
+So, any change to the `etcd` `StatefulSet` or its pods have a disruptive impact with down-time implication.
+
+One way to make such changes less disruptive could be to temporarily scale the cluster into a multi-node (3) `etcd` cluster, perform the disruptive change on a rolling manner on each of the individual nodes of the multi-node `etcd` clusters and then scale down the `etcd` back to single-instance.
+This kind of multi-step process is better implemented as a controller.
+
```

Comment:
Just a minor comment, the non-disruptive maintenance can also be implemented with two etcds and a controller. I would not invoke three etcd cluster nodes if not necessary.

Reply:
That could be possible. But as you mentioned, it would require a controller still. 2 nodes or 3.